src/Binding.lua

```diff
@@ -146,4 +205,33 @@ function Binding.create(initialValue)
 	return binding, setter
 end
 
+--[[
+	Creates a new binding which updates when any of the upstream bindings
+	updates, which can be further mapped into any value. This function will
+	be exposed to users of Roact.
+]]
+function Binding.join(bindings)
+	if config.typeChecks then
+		assert(typeof(bindings) == "table", "Bad arg #1 to Binding.join: expected table")
+
+		for key, binding in pairs(bindings) do
+			--[[
+				Bindings exclusively and always have a InternalData member
+			]]
+			assert(binding[InternalData], ("Non-binding value passed into Binding.join at index %q"):format(key))
```

Comment:
We use `InternalData` in several data structures. You should use `Type.of` for this kind of type check:
https://github.com/Roblox/roact/blob/41af74df730723ed39c02f54cae7c6c97c6c5d9c/src/createReconciler.lua#L248
```suggestion
			assert(Type.of(binding) == Type.Binding, ("Non-binding value passed into Binding.join at index %q"):format(key))
```

Reply:
I was under the assumption that a new `Symbol` was created every time. Will fix.